tests/unit/amo/components/TestCategory.js

```diff
@@ -75,163 +81,229 @@ describe('Category', () => {
-      category: fakeCategory,
-      filters: { ...filtersWithSlug, page: 1 },
-      loading: true,
-      pathname: '/themes/test/',
-      paginationQueryParams: { page: 1 },
+    const _dispatchClientApp = (clientApp) => dispatchClientMetadata({
+      store, clientApp,
     });
-  });
 
-  it('sets loading to true if categories and search are loading', () => {
-    store.dispatch(_categoriesFetch());
-    _searchStart({ filters });
-    const props = mapStateToProps(store.getState(), ownProps);
+    function _render(
+      props = {}, { autoDispatchCategories = true, ...options } = {}
+    ) {
+      if (autoDispatchCategories) {
+        _categoriesLoad({ result: [decoyCategory, targetCategory] });
+      }
+      return render(props, {
+        // Since we loaded our own, tell the parent helper not to.
+        autoDispatchCategories: false,
+        ...options,
+      });
+    }
+
+    it('looks for a category by slug, addonType, and clientApp', () => {
+      _dispatchClientApp(targetCategory.application);
+
+      const root = _render({
+        params: {
+          slug: targetCategory.slug,
+          visibleAddonType: visibleAddonType(targetCategory.type),
+        },
+      });
+
+      expect(root.find(CategoryHeader))
+        .toHaveProp('category', targetCategory);
+    });
 
-    expect(props).toMatchObject({ loading: true });
-  });
+    it('renders a 404 for unknown category', () => {
+      const root = _render({}, {
+        paramOverrides: { slug: 'unknown-category' },
+      });
 
-  it('sets loading to true if only categories are loading', () => {
-    store.dispatch(_categoriesFetch());
-    const props = mapStateToProps(store.getState(), ownProps);
+      expect(root.find(NotFound)).toHaveLength(1);
+    });
 
-    expect(props).toMatchObject({ loading: true });
-  });
+    it('renders 404 for mismatched clientApp', () => {
+      _dispatchClientApp(decoyCategory.application);
 
-  it('sets loading to true if only search is loading', () => {
-    _searchStart({ filters });
-    const props = mapStateToProps(store.getState(), ownProps);
+      const root = _render({
+        params: {
+          slug: targetCategory.slug,
+          visibleAddonType: visibleAddonType(targetCategory.type),
+        },
+      });
 
-    expect(props).toMatchObject({ loading: true });
-  });
+      expect(root.find(NotFound)).toHaveLength(1);
+    });
 
-  it('sets loading to false if nothing is loading', () => {
-    const props = mapStateToProps(store.getState(), ownProps);
+    it('renders 404 for mismatched addonType', () => {
+      _dispatchClientApp(targetCategory.application);
 
-    expect(props).toMatchObject({ loading: false });
+      const root = _render({
+        params: {
+          slug: targetCategory.slug,
+          visibleAddonType: visibleAddonType(decoyCategory.type),
+        },
+      });
+
+      expect(root.find(NotFound)).toHaveLength(1);
+    });
+
+    it('renders 404 for bogus addonType', () => {
```

Comment:
Bogus seemed like a weird word to use here but the literal definition is "not genuine or true", so I guess it's super-accurate! 😅